torch/jit/__init__.py

```diff
@@ -544,6 +555,16 @@ def _flatten(obj, params=tuple()):
     return obj_vars, obj_struct
 
 
+def _clone_inputs(args):
+    def clone_input(a):
+        if isinstance(a, Variable):
+            return Variable(a.data.clone(), requires_grad=a.requires_grad, volatile=a.volatile)
+        else:
+            return a.clone()
+    return function._nested_map(lambda o: isinstance(o, Variable) or isinstance(o, Tensor),
```

Comment:
`isinstance (o, Tensor)` doesn't work for non-default tensors. It might be better to use `torch.is_tensor` instead.
But I think we should fix this in pytorch.

Reply:
Thanks, this is a nice catch. This might be hard to fix in PyTorch because, semantically, a `Tensor` is whatever the default tensor is, and there is no intrinsic subclass relationship between it and any other tensor.

Reply:
False. With `__instancecheck__` and a few other Python tricks you can make `torch.Tensor` do all kinds of arbitrary things 😊

Reply:
Another option would be to go into the ABC thing, but I think this is not going to happen anymore https://github.com/pytorch/pytorch/issues/399?

Reply:
I prefer the ABC solution.